spec.go

```diff
@@ -12,7 +12,7 @@ type Interface interface {
 	// in the kubeconfig section of the app CR.
 	NewRESTConfigForApp(ctx context.Context, app v1alpha1.App) (*rest.Config, error)
 	// NewKubeConfigForRESTConfig returns a kubeConfig bytes for the given REST Config.
-	NewKubeConfigForRESTConfig(ctx context.Context, config *rest.Config, clusterName, namespace string) ([]byte, error)
+	NewKubeConfigForRESTConfig(config *rest.Config, clusterName, namespace string) ([]byte, error)
 	// NewRESTConfigForKubeConfig returns a REST Config for the given KubeConfigValue.
-	NewRESTConfigForKubeConfig(ctx context.Context, kubeConfig []byte) (*rest.Config, error)
```

Comment:
The consensus so far is against this change. We want to use the context as first argument in all public functions. See https://github.com/giantswarm/fmt/blob/master/go/context.md.

Reply:
@xh3b4sd Ack. This was a bad steer from me. I know we want to use context more but I'd missed it was also in `fmt`.

The problem is using the library in gsctl because its still quite different from the rest of our codebase. But the changes you're suggesting make sense to me.